includes/class-newspack-popups-parse-logs.php

```diff
@@ -0,0 +1,176 @@
+<?php
+/**
+ * Newspack Segmentation Plugin
+ *
+ * @package Newspack
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+require_once dirname( __FILE__ ) . '/../api/segmentation/class-segmentation.php';
+
+/**
+ * Main Newspack Segmentation Plugin Class.
+ */
+final class Newspack_Popups_Parse_Logs {
+	/**
+	 * The single instance of the class.
+	 *
+	 * @var Newspack_Popups_Parse_Logs
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Main Newspack Segmentation Plugin Instance.
+	 * Ensures only one instance of Newspack Segmentation Plugin Instance is loaded or can be loaded.
+	 *
+	 * @return Newspack Segmentation Plugin Instance - Main instance.
+	 */
+	public static function instance() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_filter( 'cron_schedules', [ __CLASS__, 'add_cron_interval' ] ); // phpcs:ignore WordPress.WP.CronInterval.CronSchedulesInterval
+		add_action( 'newspack_popups_segmentation_cron_hook', [ __CLASS__, 'parse_events_logs' ] );
+		if ( ! wp_next_scheduled( 'newspack_popups_segmentation_cron_hook' ) ) {
+			wp_schedule_event( time(), 'every_minute', 'newspack_popups_segmentation_cron_hook' );
+		}
+	}
+
+
+	/**
+	 * Add a CRON interval of one minute.
+	 *
+	 * @param object $schedules The schedules.
+	 */
+	public static function add_cron_interval( $schedules ) {
+		$schedules['every_minute'] = [
+			'interval' => 60,
+			'display'  => esc_html__( 'Every Minute' ),
+		];
+		return $schedules;
+	}
+
+	/**
+	 * Insert multiple rows in one DB transaction.
+	 *
+	 * @param string $table_name Table name.
+	 * @param array  $rows Rows to insert.
+	 * @param array  $column_names Names of the columns.
+	 * @param array  $placeholder Row placeholder for wpdb->prepare (e.g. `(%s, %s)`).
+	 */
+	public static function bulk_db_insert( $table_name, $rows, $column_names, $placeholder ) {
+		if ( 0 === count( $rows ) ) {
+			return;
+		}
+
+		global $wpdb;
+
+		$column_names = implode( ', ', $column_names );
+		$query        = "INSERT INTO $table_name ($column_names) VALUES ";
```

Comment:
According to this SO post, we can insert `13107` rows (there are 5 placeholders per row). With cron running every 15 minutes, that gives us ~14 events per second. With one event per pageview, our upper limit is about `1.2M` PVs per day (with cron running every 5 minutes, it's ~43 events, `3.7M` PVs/day). 
Note that not all pageviews will log events (only articles), so that gives us an even bigger margin of error. If these calculations are correct, I think we're safe.

Reply:
Sounds good. Perhaps as a later enhancement we can insert them in groups of 10k max to be ultra-safe. Agreed we're safe for now.